src/client/index.ts

```diff
@@ -283,23 +302,57 @@ export class Resend {
 
     if (this.config.apiKey === "") throw new Error("API key is not set");
 
-    const id = await ctx.runMutation(this.component.lib.sendEmail, {
-      options: await configToRuntimeConfig(this.config, this.onEmailEvent),
-      ...sendEmailArgs,
-      to:
-        typeof sendEmailArgs.to === "string"
-          ? [sendEmailArgs.to]
-          : sendEmailArgs.to,
-      cc: toArray(sendEmailArgs.cc),
-      bcc: toArray(sendEmailArgs.bcc),
-    });
-
-    return id as EmailId;
+    // Prepare the mutation args based on whether it's a template or traditional email
+    if ("template" in sendEmailArgs) {
+      // Template-based email
+      const id = await ctx.runMutation(this.component.lib.sendEmail, {
+        options: await configToRuntimeConfig(this.config, this.onEmailEvent),
+        from: sendEmailArgs.from,
+        to:
+          typeof sendEmailArgs.to === "string"
+            ? [sendEmailArgs.to]
+            : sendEmailArgs.to,
+        cc: toArray(sendEmailArgs.cc),
+        bcc: toArray(sendEmailArgs.bcc),
+        subject: sendEmailArgs.subject,
+        replyTo: sendEmailArgs.replyTo,
+        headers: sendEmailArgs.headers,
+        templateId: sendEmailArgs.template.id,
+        templateVariables: JSON.stringify(sendEmailArgs.template.variables),
+      });
+      return id as EmailId;
+    } else {
+      // Traditional email
+      const id = await ctx.runMutation(this.component.lib.sendEmail, {
+        options: await configToRuntimeConfig(this.config, this.onEmailEvent),
+        from: sendEmailArgs.from,
+        to:
+          typeof sendEmailArgs.to === "string"
+            ? [sendEmailArgs.to]
+            : sendEmailArgs.to,
+        cc: toArray(sendEmailArgs.cc),
+        bcc: toArray(sendEmailArgs.bcc),
+        replyTo: sendEmailArgs.replyTo,
+        headers: sendEmailArgs.headers,
+        subject: sendEmailArgs.subject,
+        html: sendEmailArgs.html,
+        text: sendEmailArgs.text,
+      });
+      return id as EmailId;
+    }
```

Comment:
```suggestion
    options: await configToRuntimeConfig(this.config, this.onEmailEvent),
      ...sendEmailArgs,
      to:
        typeof sendEmailArgs.to === "string"
          ? [sendEmailArgs.to]
          : sendEmailArgs.to,
      cc: toArray(sendEmailArgs.cc),
      bcc: toArray(sendEmailArgs.bcc),
    });

    return id as EmailId;
```
the only risk here is that passing extra args will fail -but in some ways that's a good thing